scripts/hook.py

```diff
@@ -600,8 +606,8 @@ def forward(self, x, timesteps=None, context=None, **kwargs):
             h = aligned_adding(h, total_controlnet_embedding.pop(), require_inpaint_hijack)
 
             # U-Net Decoder
-            for i, module in enumerate(self.output_blocks):
-                h = th.cat([h, aligned_adding(hs.pop(), total_controlnet_embedding.pop(), require_inpaint_hijack)], dim=1)
+            for module, (hs_item, controlnet_embedding_item) in zip(self.output_blocks, reversed(list(zip(hs, total_controlnet_embedding)))):
```

Comment:
`hs` has 9 tensor elements, while `total_controlnet_embedding` has 10 tensor elements and 2 `0` padding at the end.

This change aligns the 2 arrays, but I am not sure it is the correct fix here.

Reply:
That impl does not handle a tailing 0 for middle block weights. Revised the impl to calculate array length based on model type.